### PR TITLE
Correct locking on LED thread

### PR DESCRIPTION
**Pull request by andyp1per**
The new LED thread had a locking problem which could lead to watchdogs

### CODE REVIEW

File: libraries/AP_HAL_ChibiOS/RCOutput.cpp [hunk 1]
```diff
@@ -1608,11 +1608,14 @@ bool RCOutput::serial_led_send(pwm_group &group)
     }
 
 #ifndef DISABLE_DSHOT
-    if (irq.waiter || !group.dma_handle->lock_nonblock()) {
-        // doing serial output, don't send Serial LED pulses
+    if (irq.waiter || (group.dshot_state != DshotState::IDLE && group.dshot_state != DshotState::RECV_COMPLETE)) {
```
Comment (tridge) [inline libraries/AP_HAL_ChibiOS/RCOutput.cpp:1611]: LED don't do receive .... this check not needed

File: libraries/AP_HAL_ChibiOS/RCOutput.cpp [hunk 2]
```diff
@@ -288,7 +288,7 @@ void RCOutput::dshot_collect_dma_locks(uint64_t time_out_us, bool led_thread)
     for (int8_t i = NUM_GROUPS - 1; i >= 0; i--) {
         pwm_group &group = pwm_group_list[i];
 
-        if ((led_thread && !is_led_protocol(group.current_mode)) || is_led_protocol(group.current_mode)) {
+        if (led_thread != is_led_protocol(group.current_mode)) {
```
Comment (tridge) [inline libraries/AP_HAL_ChibiOS/RCOutput.cpp:291]: this looks like a non functional change

Comment (andyp1per) [reply libraries/AP_HAL_ChibiOS/RCOutput.cpp:291]: If led_thread is true and is_led_protocol(group.current_mode) is true then we will skip collecting locks, so we are not collecting locks in the led thread

**Review by tridge — APPROVED** [review]

**Comment by rmackay9** [comment]
I've added the 4.4 projects to this PR because they were missing but this is included in beta2.

**Pull request merged**